File: python練習/GeometricObject_Inheritance/Circle.py
import math
from GeometricObject import GeometricObject

class Circle(GeometricObject):
    def __init__(self, radius, color, filled):
        super().__init__(color, filled)
        self.__radius = radius
        
    def setRadius(self, radius):
        self.__radius = radius
        
    def getRadius(self):
        return self.__radius
    
    def getDiameter(self):
        return int(self.__radius) * 2
    
    def getPerimeter(self):
        return 2 * math.pi * int(self.__radius)
    
    def getArea(self):
        return math.pi * math.pow(int(self.__radius), 2)

File: 第四次作業/Circle.cpp
#include "Circle.h"
#include <math.h>
#include <sstream>
#include <string>
using namespace std;

Circle::Circle()
{

}

Circle::Circle(double radius, string color, bool filled)
{
	this->radius = radius;
	setColor(color);
	setFilled(filled);
}

void Circle::setRadius(double radius)
{
	this->radius = radius;
}

double Circle::getRadius() const
{
	return radius;
}

double Circle::getPerimeter() const
{
	return 2 * 3.14159 * radius;
}

double Circle::getArea() const
{
	return 3.14159 * pow(radius, 2);
}

string Circle::toString() const
{
	stringstream ss;
	ss << "circle, " << radius << ", " << getColor();
	return ss.str();
}